app/models/spree/payment/processing_decorator.rb

```diff
@@ -0,0 +1,11 @@
+Spree::Payment::Processing.module_eval do
+  def cancel!
```

Comment:
Yeah like Jordan suggests it would be great if we could avoid this, one idea that comes to mind is using the `response_code`, I believe the response code is recorded on the payment, so we should be able to fetch the payment using it, from there we can get the amazon_transaction using `payment.source` and the order using `payment.order` :)